devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+	log.debug "BasicSet: ${cmd}"
+	def input1SwitchType = Integer.parseInt(state.currentPreferencesState.input1SwitchType.value)
+
+	if(input1SwitchType == INPUT_TYPE_POTENTIOMETER) {
+		log.debug "BasicSet: ${cmd} / INPUT_TYPE_POTENTfIOMETER"
+		[response(zwave.switchMultilevelV3.switchMultilevelGet().format())]
+	} else if (input1SwitchType == INPUT_TYPE_BI_STABLE_SWITCH) {
+		log.debug "BasicSet: ${cmd} / INPUT_TYPE_BI_STABLE_SWITCH"
+		dimmerEvents(cmd)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd, ep = null) {
+	log.debug "SwitchMultilevelReport: ${cmd}"
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd, ep = null) {
+	log.debug "SwitchMultilevelSet: ${cmd}"
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep = null) {
+	log.debug "MeterReport: ${cmd}"
+	handleMeterReport(cmd)
+}
+
+def handleMeterReport(cmd) {
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			log.debug("createEvent energy")
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+		} else if (cmd.scale == 1) {
+			log.debug("createEvent energy kVAh")
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
+		} else if (cmd.scale == 2) {
+			log.debug("createEvent power")
+			createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd, ep = null) {
+	createDimmerEvents(cmd.value)
+}
+
+private createDimmerEvents(cmdValue){
+	def value = (cmdValue ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value)]
+	if (cmdValue && cmdValue <= 100) {
+		result << createEvent(name: "level", value: cmdValue == 99 ? 100 : cmdValue)
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd, ep = null) {
+	log.info "SensorMultilevelReport: ${cmd}, endpoint: ${ep}"
+	def result = []
+
+	def map = [:]
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	handleChildEvent(map)
+	result << createEvent(map)
+}
+
+def handleChildEvent(map) {
+	def childDni = "${device.deviceNetworkId}:" + 2
+	log.debug "handleChildEvent / find child device: ${childDni}"
+	def childDevice = childDevices.find { it.deviceNetworkId == childDni }
+
+	if(!childDevice) {
+		log.debug "handleChildEvent / creating a child device"
+		def childComponentLabel	= "Qubino Temperature Sensor"
+		def childDthName		= "Child Temperature Sensor"
+		def childDthNamespace 	= "qubino"
```

Comment:
if these are always the same, just inline them. You can create constants for them if you want to avoid magic strings.

Reply:
Good point. Fixed.